File: src/main/java/zw/co/hisolutions/core/service/VendorService.java
package zw.co.hisolutions.core.service;
 
import zw.co.hisolutions.core.entity.Vendor;
import zw.co.hisolutions.common.service.GenericService;

public interface VendorService extends GenericService<Vendor, Long>{
    public Vendor findByName(String name);   
}

File: frontend/src/app/admin/services/rest/vendor.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';   

import {RestDataService} from 'shared/services/rest-data.service'; 

@Injectable()
export class  VendorService extends RestDataService {

    constructor(httpClient: HttpClient) {
        const resourcePath =  '/rest/vendors'; 
        super(httpClient, resourcePath);
    }

}
